src/core/shape/2d_primitives.js

```diff
@@ -518,7 +522,7 @@ p5.prototype.quad = function(...args) {
  * @param  {Number} x  x-coordinate of the rectangle.
  * @param  {Number} y  y-coordinate of the rectangle.
  * @param  {Number} w  width of the rectangle.
- * @param  {Number} [h]  height of the rectangle.
+ * @param  {Number} h  height of the rectangle.
```

Comment:
Hey @lmccart , i need some advice over here. I tested the `rect` function and on not providing the 4th argument, the rectangle is not rendered on the canvas and a error messages pops up in the console. But here it was documented as optional, so i changed it to match the current behaviour of the code. Now the tests are failing due to this change. Should i revert back this change or update the test ? I am not sure what i am missing.

Reply:
I think reverting this commit should handle all of the changes. https://github.com/processing/p5.js/commit/98241d6c02c67cacfbe09c8cccfd1bc9356b3cdc#diff-27b6dc80741d2191e0ca1b5030cf73e1

Reply:
The code should probably stay the same, it is a documented feature (passing three arguments) and changing it would be considered a breaking change. Maybe look into why FES is not allowing the 3 argument version, maybe the signature isn't registered correctly or something.

Reply:
Well where did ya test rect() function with 3 arguments? If on web editor then it uses 0.10.2 version of p5.js by default and it works perfectly fine with version 1.0.0 of p5.js . so I think changes for rect() to take 3 arguments were made after release of 0.10.2 version , and maybe that's the reason. After reading source code and testing it on version 1.0.0 I didn't found any visible errors so I think you should revert it back to [h]

Check this example out,
https://editor.p5js.org/DivyamAhuja/sketches/z_PtePeH

Reply:
@limzykenneth he updated signature, changing h from optional to required so now unit test case with 3 input is expecting 4 input parameters and is throwing error

Reply:
OK, got the error. Yes, i had tested it on the web editor. @DivyamAhuja, you are correct. I will push the necessary changes as well. Thanks for your help @DivyamAhuja , @limzykenneth.